src/components/product-modal/ProductModalContainer.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+} from "@chakra-ui/react";
+import Product from "./Product";
+import { getProductById } from "../../services/products";
+import {
+  isChildProductResource,
+  isSimpleProductResource,
+} from "../../services/helper";
+import {
+  retrieveBaseProps,
+  retrieveChildProps,
+  retrieveSimpleProps,
+} from "../../lib/retrieve-product-props";
+import { useEffect, useState } from "react";
+import { IProduct } from "../../lib/product-types";
+import { GetStaticPropsResult } from "next/types";
+
+interface ProductModalProps {
+  productId: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const ProductModalContainer = ({
+  productId,
+  isOpen,
+  onClose,
+}: ProductModalProps) => {
+  const [productProps, setProductProps] =
+    useState<GetStaticPropsResult<IProduct>>();
+
+  const onSkuIdChange = (id: string) => {
+    fetchProduct(id);
+  };
+
+  const fetchProduct = async (id: string) => {
+    const product = await getProductById(id);
+    const productData = product.data;
+    const retrievedResults = isSimpleProductResource(productData)
+      ? retrieveSimpleProps(product)
+      : isChildProductResource(productData)
+      ? await retrieveChildProps(product)
+      : await retrieveBaseProps(product);
+    setProductProps(retrievedResults);
+  };
+
+  useEffect(() => {
+    isOpen && fetchProduct(productId);
+  }, [productId, isOpen]);
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent maxW="90%" w="900px">
```

Comment:
could having a fixed px width cause problems anywhere on smaller screens?

Reply:
<img width="360" alt="image" src="https://user-images.githubusercontent.com/33585095/198053506-f74fa2f2-0901-49cd-adb6-9e910642a253.png">
looks good to me. I think `maxW` is always respected
btw, chakra UI Modal has the size prop but even the biggest value is too small for us